app/main.py

```diff
@@ -1,2 +1,71 @@
+from typing import Hashable, Any
+from dataclasses import dataclass
+
+INITIAL_CAPACITY = 8
+
+
+@dataclass
+class Node:
+    key: Hashable
+    dict_hash: int
+    value: Any
+
+
 class Dictionary:
-    pass
+    def __init__(self, capacity: int = INITIAL_CAPACITY) -> None:
+        self.capacity = capacity
+        self.length = 0
+        self.threshold = 2 / 3
+        self.hash_table: list[Node | None] = [None] * self.capacity
+
+    def __getitem__(self, key: Hashable) -> Any:
+        index = self._calculation_index(key)
+
+        if self.hash_table[index] is None:
+            raise KeyError(f"Key {key} doesn't exist in dictionary")
+
+        return self.hash_table[index].value
+
+    def __setitem__(self, key: Any, value: Any) -> None:
+        if self.length / self.capacity > self.threshold:
+            self._resize_hash_table()
+
+        index = self._calculation_index(key)
+
+        if (
+                self.hash_table[index] is not None
+                and self.hash_table[index].key == key
+        ):
+            self.hash_table[index].value = value
+        else:
+            self.length += 1
+            self.hash_table[index] = Node(key, hash(key), value)
```

Comment:
The `Node` class includes a `dict_hash` field, but it is not used in the current implementation. Consider using it to store the precomputed hash value of the key to optimize hash calculations.